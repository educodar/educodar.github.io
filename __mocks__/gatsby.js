const React = require("react")
const gatsby = jest.requireActual("gatsby")

const gatsbyMock = {
  ...gatsby,
  graphql: jest.fn(),
  Link: jest
    .fn()
    .mockImplementation(
      ({
        activeClassName,
        activeStyle,
        getProps,
        innerRef,
        partiallyActive,
        ref,
        replace,
        to,
        ...rest
      }) =>
        React.createElement("a", {
          ...rest,
          href: to,
        })
    ),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn(),
}

gatsbyMock.StaticQuery.mockImplementation(({ render }) =>
  render({
    site: {
      siteMetadata: {
        title: "Educodar - Aulas de programação grátis para jovens",
        titleTemplate: "%s · Educodar",
        siteUrl: "https://educodar.com.br",
        description:
          "Projeto de ensino de programação básica e exposição a T.I. para estudantes de escolas de comunidades. Iniciativa sem fins lucrativos. Cursos e alimentação gratuitos.",
        image: "/images/logo.png",
      },
    },
  })
)

module.exports = gatsbyMock
