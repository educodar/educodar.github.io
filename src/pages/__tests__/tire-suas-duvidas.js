import React from "react"
import renderer from "react-test-renderer"
import { StaticQuery } from "gatsby"

import FAQPage from "../tire-suas-duvidas"

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({ site: { siteMetadata: {} } })
  )
})

describe("FAQ page", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<FAQPage />)
    expect(tree).toMatchSnapshot()
  })
})
