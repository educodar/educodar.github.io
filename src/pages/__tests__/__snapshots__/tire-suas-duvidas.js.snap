// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQ page renders correctly 1`] = `
Array [
  <nav
    itemProp="breadcrumb"
  >
    <a
      href="/"
    >
      Página inicial
    </a>
    <span
      aria-current="page"
    >
      Tire suas dúvidas
    </span>
  </nav>,
  <article>
    <h1>
      Tire suas dúvidas
    </h1>
    <dl>
      <div
        itemProp="mainEntity"
        itemScope={true}
        itemType="https://schema.org/Question"
      >
        <dt
          itemProp="name"
        >
          O que é Educodar?
        </dt>
        <dd
          itemProp="acceptedAnswer"
          itemScope={true}
          itemType="https://schema.org/Answer"
        >
          <div
            itemProp="text"
          >
            É um projeto voluntário que…
          </div>
        </dd>
      </div>
      <div
        itemProp="mainEntity"
        itemScope={true}
        itemType="https://schema.org/Question"
      >
        <dt
          itemProp="name"
        >
          O Educodar é um curso?
        </dt>
        <dd
          itemProp="acceptedAnswer"
          itemScope={true}
          itemType="https://schema.org/Answer"
        >
          <div
            itemProp="text"
          >
            Não.
          </div>
        </dd>
      </div>
      <div
        itemProp="mainEntity"
        itemScope={true}
        itemType="https://schema.org/Question"
      >
        <dt
          itemProp="name"
        >
          Tenho que pagar para estudar no Educodar?
        </dt>
        <dd
          itemProp="acceptedAnswer"
          itemScope={true}
          itemType="https://schema.org/Answer"
        >
          <div
            itemProp="text"
          >
            Não, ele é 100% gratuito.
          </div>
        </dd>
      </div>
      <div
        itemProp="mainEntity"
        itemScope={true}
        itemType="https://schema.org/Question"
      >
        <dt
          itemProp="name"
        >
          Como o Educodar me ajuda a ingressar no mercado de trabalho?
        </dt>
        <dd
          itemProp="acceptedAnswer"
          itemScope={true}
          itemType="https://schema.org/Answer"
        >
          <div
            itemProp="text"
          >
            <p>
              Se você for menor de idade e, portanto, ainda estiver na época da escola, nossa preocupação não será em arrumar um emprego pra você. Nós vamos mostrar a você que a área de tecnologia é uma possível oportunidade para o seu futuro.
            </p>
            <p>
              Se você é maior de idade e está procurando uma oportunidade de trabalho, não podemos garantir que você saia daqui com um emprego. Porém, vamos te apoiar e te dar direcionamento em relação a currículo, processos seletivos e possíveis cursos.
            </p>
          </div>
        </dd>
      </div>
      <div
        itemProp="mainEntity"
        itemScope={true}
        itemType="https://schema.org/Question"
      >
        <dt
          itemProp="name"
        >
          Tenho que fazer provas e tirar nota boa?
        </dt>
        <dd
          itemProp="acceptedAnswer"
          itemScope={true}
          itemType="https://schema.org/Answer"
        >
          <div
            itemProp="text"
          >
            Não. Pra gente, o mais importante é que todas as pessoas aprendam. Seu desempenho será acompanhado durante as aulas, e sempre vai ter alguém para te dar apoio se você tiver alguma dúvida.
          </div>
        </dd>
      </div>
      <div
        itemProp="mainEntity"
        itemScope={true}
        itemType="https://schema.org/Question"
      >
        <dt
          itemProp="name"
        >
          Como funciona o Educodar?
        </dt>
        <dd
          itemProp="acceptedAnswer"
          itemScope={true}
          itemType="https://schema.org/Answer"
        >
          <div
            itemProp="text"
          >
            Nós dividimos o Educodar em duas fases. Na primeira, fazemos um
            <em
              lang="en"
            >
              workshop
            </em>
             super divertido, que vai ajudar você a aprender conceitos básicos de programação (como lógica de programação, por exemplo). Na segunda fase, você aprenderá a programar.
          </div>
        </dd>
      </div>
      <div
        itemProp="mainEntity"
        itemScope={true}
        itemType="https://schema.org/Question"
      >
        <dt
          itemProp="name"
        >
          O que aprenderei no Educodar?
        </dt>
        <dd
          itemProp="acceptedAnswer"
          itemScope={true}
          itemType="https://schema.org/Answer"
        >
          <div
            itemProp="text"
          >
            No início, você aprenderá 3 linguagens que são utilizadas para construir sites e até alguns aplicativos. Elas são o HTML, CSS e JavaScript. Se você quiser saber mais sobre essas linguagens, pode acessar…
          </div>
        </dd>
      </div>
      <div
        itemProp="mainEntity"
        itemScope={true}
        itemType="https://schema.org/Question"
      >
        <dt
          itemProp="name"
        >
          Não sei nada de informática ou programação. Posso me candidatar?
        </dt>
        <dd
          itemProp="acceptedAnswer"
          itemScope={true}
          itemType="https://schema.org/Answer"
        >
          <div
            itemProp="text"
          >
            Sim. Todas as pessoas são bem-vindas.
          </div>
        </dd>
      </div>
      <div
        itemProp="mainEntity"
        itemScope={true}
        itemType="https://schema.org/Question"
      >
        <dt
          itemProp="name"
        >
          Eu não tenho um computador. Posso participar?
        </dt>
        <dd
          itemProp="acceptedAnswer"
          itemScope={true}
          itemType="https://schema.org/Answer"
        >
          <div
            itemProp="text"
          >
            Sim. Enquanto você estiver em nosso local de aprendizado, você poderá usar um computador que ficará reservado para você. Porém, por questões de segurança, nós não podemos emprestar o computador para você levar para casa.
          </div>
        </dd>
      </div>
      <div
        itemProp="mainEntity"
        itemScope={true}
        itemType="https://schema.org/Question"
      >
        <dt
          itemProp="name"
        >
          O Educodar é um processo de recrutamento?
        </dt>
        <dd
          itemProp="acceptedAnswer"
          itemScope={true}
          itemType="https://schema.org/Answer"
        >
          <div
            itemProp="text"
          >
            Não. O programa Educodar não tem vínculo…
          </div>
        </dd>
      </div>
    </dl>
  </article>,
]
`;
